main/src/library/Array.flix

```diff
@@ -270,13 +292,15 @@ namespace Array {
     ///
     /// The result is a new array.
     ///
+    @Time(time(f) * length(a)) @Space(space(f) * length(a))
     pub def map(f: a -> b & e, a: Array[a]): Array[b] & Impure =
         let len = length(a);
         init(i -> f(a[i]), len)
 
     ///
     /// Apply `f` to every element in array `a`. Array `a` is mutated.
     ///
+    @Time(time(f) * length(a)) @Space(space(f))
```

Comment:
I think the space needs to be multiplied by the length

Reply:
My reasoning is that if every element is replaced by a potentially much larger element from f.

Reply:
If you have an array of empty strings and replace every element by a very large string then you know have an array of large strings.